### PR TITLE
Implement reshape_as

### LINKED ISSUE #9416: [feature request] reshape_as
we have view_as and expand_as

### PR DESCRIPTION
**Pull request by vishwakftw**
1. Added tests
2. Added doc string
3. Remove view_as redundant definition from tensor.py

Closes #9416

### CODE REVIEW

File: torch/_tensor_docs.py [hunk 1]
```diff
@@ -1666,6 +1666,18 @@ def callable(a, b) -> number
 See :func:`torch.reshape`
 """)
 
+add_docstr_all('reshape_as',
+               r"""
+reshape_as(other) -> Tensor
+
+Returns this tensor as the same shape as :attr:`other`.
+``self.reshape_as(other)`` is equivalent to ``self.reshape(other.shape)``.
```
Comment (SsnL) [inline torch/_tensor_docs.py]: nit: use `other.size()` to be consistent with `view_as`

File: torch/_tensor_docs.py [hunk 2]
```diff
@@ -2407,6 +2419,18 @@ def callable(a, b) -> number
 
 """)
 
+add_docstr_all('view_as',
+               r"""
+view_as(other) -> Tensor
+
+View this tensor as the same size as :attr:`other`.
```
Comment (SsnL) [inline torch/_tensor_docs.py]: Can you add a sentence that links to `` :meth:`torch.Tensor.view` `` on the constraints of `view`?

**Comment by SsnL** [comment]
Looks great in general!

**Comment by SsnL** [comment]
Thanks @vishwakftw !

**Review by ssnl — APPROVED** [review]

**Comment by ezyang** [comment]
CI failures look fake

**Review by ezyang — APPROVED** [review]

**Pull request closed without merging**